### DISCUSSION TITLE
How does the Google Drive integration work?

### DISCUSSION THREAD

**Discussion by watermosaic** [gildas-lormeau/SingleFile#693] [Q&A]
I'm trying to activate the Google Drive integration. The instructions in the readme are sparse:

> Enable the option "Misc. > save to Google Drive" to upload pages to Google Drive.

I've tried this, and also set `forceWebAuthFlow` to `true`, but nothing happened. I logged into Chrome and restarted the browser and retried the options, and got no Web Auth flow or anything. 

Can someone provide more detailed instructions for how to make this work, or let me know if I should see another solution? I'm trying to autodownload to Google Drive and then read directly into an AWS pipeline. 

(Currently I can only get autodownload working in Firefox, so it would be fantastic if the Drive integration can also be implemented there.)

Thank you :)

**Comment by gildas-lormeau** [comment]
Did you try to save a page? Nothing will happen if you don't save any page.

**Reply by watermosaic** [reply]
Hi @gildas-lormeau This works in Chrome, but doesn't seem to work in Firefox. I can't get the autosave working in Chrome, and that's the most important feature to me, because of the bulk of saving that I'm doing.

**Reply by gildas-lormeau** [reply]
I cannot reproduce the issue in Firefox, the auto-save to Google Drive works fine on my side. Can you reset the options, change the `Destination` option to `upload to Google Drive` and do the test again? You should not need to change the `forceWebAuthFlow` value, it's already set to `true` by default in Firefox.

**Reply by neurokrish** [reply]
Hi, I seem to have a similar issue. I am using Vivaldi as my browser. When I choose save to Google drive in SingleFile, on saving a page, it says that the user disable signin to Google Drive in the browser. I am not sure how to fix this or where to enable this in Vivaldi. Is there a way around this?

**Reply by gildas-lormeau** [reply]
@neurokrish That's a weird issue, maybe it's specific to Vivaldi? You can still try to set `forceWebAuthFlow` to `true`, it's a hidden setting. See https://github.com/gildas-lormeau/SingleFile/wiki/Hidden-options for more information about how to update it.

**Reply by neurokrish** [reply]
Thanks @gildas-lormeau . Changing the `forceWebAuthFlow` to `true`  worked. Looks like it is a Vivaldi browser issue then.

**Reply by gildas-lormeau** [reply]
Thank you for the feedback @neurokrish, I'll try to see if I can automate this.
